File: Examples/cosmology/fsigma8.cpp
// ============================================
// Example code: how to estimate f(z)*sigma8(z) 
// ============================================

#include "LCDM.h"
#include "RSD.h"

using namespace std;

int main () {

  try {
  
    // --------------------------------------------------------
    // ---------------- set the cosmological model ------------
    // --------------------------------------------------------
    
    auto cosmology = make_shared<cbl::cosmology::LCDM>("Planck18");
    

    // ------------------------------------------------------
    // ---------------- estimate f*sigma8 at z=1 ------------
    // ------------------------------------------------------

    cbl::cosmology::RSD rsd(cosmology);
    
    const double redshift = 1.;
    const string method = "EisensteinHu";

    const double fs8 = rsd.fsigma8(redshift, method);

    cout << "f*sigma8(z=" << redshift << ") = " << fs8 << endl;

    cbl::Beep("the linear growth rate times sigma8, at redshift "+cbl::conv(redshift, cbl::par::fDP0)+", is equal to "+cbl::conv(fs8, cbl::par::fDP1));
    
  }

  catch(cbl::glob::Exception &exc) { cerr << exc.what() << endl; exit(1); }
  
  return 0;
} 

File: Examples/cosmology/fsigma8.py
## ============================================
## Example code: how to estimate f(z)*sigma8(z) 
## ============================================

# to ensure compatibility in Python versions 2.x and 3.x
from __future__ import print_function

# import the CosmoBolognaLib modules 
import cblCosmology as cbl

# set the cosmological model, using default parameters
cosmology = cbl.LCDM("Planck18")

# set the object to manage redshift-space distortion computations
rsd = cbl.RSD(cosmology)

# estimate f*sigma8 at z=1 with CAMB
print("f*sigma8(z=1) =", rsd.fsigma8(1., "EisensteinHu"))
